diff --git a/includes/menu.inc b/includes/menu.inc
index 2be0903..6e90e00 100644
--- a/includes/menu.inc
+++ b/includes/menu.inc
@@ -2486,6 +2486,7 @@ function menu_link_get_preferred($path = NULL, $selected_menu = NULL) {
     // Weight must be taken from {menu_links}, not {menu_router}.
     $query->addField('ml', 'weight', 'link_weight');
     $query->fields('m');
+    $query->orderBy('ml.customized', 'DESC');
     $query->condition('ml.link_path', $path_candidates, 'IN');
 
     // Sort candidates by link path and menu name.
