diff --git entity/theme/entity.theme.inc entity/theme/entity.theme.inc
index d365996..eef0df8 100644
--- entity/theme/entity.theme.inc
+++ entity/theme/entity.theme.inc
@@ -169,7 +169,7 @@ function template_preprocess_entity(&$variables) {
   $variables['title'] = check_plain(entity_label($entity_type, $entity));
 
   $uri = entity_uri($entity_type, $entity);
-  $variables['url'] = $uri ? url($uri['path'], $uri['options']) : FALSE;
+  $variables['url'] = ($uri && !empty($uri['path'])) ? url($uri['path'], $uri['options']) : FALSE;
 
   if (isset($variables['elements']['#page'])) {
     // If set by the caller, respect the page property.
@@ -177,7 +177,7 @@ function template_preprocess_entity(&$variables) {
   }
   else {
     // Else, try to automatically detect it.
-    $variables['page'] = $uri && $uri['path'] == $_GET['q'];
+    $variables['page'] = $uri && !empty($uri['path']) && $uri['path'] == $_GET['q'];
   }
 
   // Helpful $content variable for templates.
